docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md

```diff
@@ -0,0 +1,139 @@
+
+![](images/eq2.png)
+
+
+L denotes cross entropy or other loss function. The first term in the formula corresponds to the output of the optimized expansion, which pushes the samples into the corresponding class groups to maintain accurate identification in the closed set. In the second term, matching the task to K+1 classes makes the virtual classifier output the second highest probability, tries to associate the position of the virtual classifier in the center space, and controls the distance to the virtual classifier as the second closest distance among all class centers. Thus, it seeks a trade-off between correctly classifying closed set instances and retaining the probability of a new class as a classifier placeholder. During training, it can be positioned between target and non-target classes. In the case of novel classes, the predictive power of the virtual classifier can be high because all known classes are non-target classes. Therefore, it is considered as an instance-related threshold that can be well adapted to each known class.
+
+
+
+##### Learning data placeholders
+
+The purpose of learning data placeholders is to transform closed-set training into open-set training. The combined data placeholders should have two main characteristics, the distribution of these samples should look novel and the generation process should be fast. In this paper, we simulate new patterns with multiple mixtures. Equation 6 in the paper gives two samples from different categories and mixes them in the middle layer.
+
+![](images/eq3.png)
+
+The results of the mixture are passed through the later layers to obtain the new model embedding post. considering that the interpolation between two different clusters is usually a low confidence prediction region. The paper treats the embedding φpost (̃xpre) as an embedding of the open set class and trains it as a new class.
+
+![](images/eq4.png)
+
+It is clear that the formulation in the paper does not consume additional time complexity, which would generate new situations between multiple decision boundaries. In addition, streamwise blending allows better use of interpolation of deeper hidden representations to generate new patterns in the improved embedding space, which better represents the new distribution. As illustrated in the figure above, the blending instance pushes the decision boundaries in the embedding space to two separate locations of the classes. With the help of the data placeholders, the embedding of the known classes will be tighter, leaving more places for the new classes.
+
+
+
+**PROSER algorithm training process**
+
+![](images/process.png)
+
+### Embedded in Ianvs
+
+![ ](images/ianvs_arch.png)
+
+After training, the unknown task algorithm model is placed in the Lifelong Learning Paradigm section of the Test Case Controller module as one of the algorithms for unknown task identification in the lifelong learning paradigm.
```

Comment:
Good to learn that. It is true that the algorithm should put in the lifelong learning lib in the test case controller. But currently in this proposal the ianvs lifelong learning procedure is not clear. How about put the ianvs lifelong modules in a figure?

Reply:
We see the ianvs arch but the relationship between the proposal and ianvs can be more straightforward. My suggestion is to add a figure showing the proposal module in the overall lifelong learning flow to see how it works.

Reply:
still need to follow up. highlight the proposed module in overall flow.

Reply:
Completed~